CMakeLists.txt

```diff
@@ -0,0 +1,336 @@
+	)
+
+	if(32bit)
+		FIND_LIBRARY(FRIBIDI_LIBRARY libfribidi
+		PATH
+		${PROJECT_SOURCE_DIR}/win/lib/windows32
+		)
+	else()
+		FIND_LIBRARY(FRIBIDI_LIBRARY libfribidi
+		PATH
+		${PROJECT_SOURCE_DIR}/win/lib/windows64
+		)
+	endif(32bit)
+
+	message("-- Found Fribidi: ${FRIBIDI_LIBRARY} ")
+
+else()
+	include(CheckFunctionExists)
+	SET(FRIBIDI_FOUND "NO")
+
+	FIND_PATH(FRIBIDI_INCLUDE_DIR fribidi/fribidi.h
+		/usr/local/include
+		/usr/include
+	)
+
+	SET(FRIBIDI_NAMES ${FRIBIDI_NAMES} fribidi libfribidi)
+	FIND_LIBRARY(FRIBIDI_LIBRARY
+		NAMES ${FRIBIDI_NAMES}
+		PATHS /usr/lib /usr/local/lib
+	)
+
+	IF (FRIBIDI_LIBRARY AND FRIBIDI_INCLUDE_DIR)
+		SET(CMAKE_REQUIRED_INCLUDES ${FRIBIDI_INCLUDE_DIR})
+		SET(CMAKE_REQUIRED_LIBRARIES ${FRIBIDI_LIBRARY})
+		CHECK_FUNCTION_EXISTS(fribidi_utf8_to_unicode FOUND_fribidi_utf8_to_unicode)
+		IF(FOUND_fribidi_utf8_to_unicode)
+			SET(FRIBIDI_LIBRARIES ${FRIBIDI_LIBRARY})
+			SET(FRIBIDI_FOUND "YES")
+		ELSE(FOUND_fribidi_utf8_to_unicode)
+			SET(FRIBIDI_LIBRARIES "NOTFOUND")
+			SET(FRIBIDI_INCLUDE_DIR "NOTFOUND")
+			SET(FRIBIDI_FOUND "NO")
+		ENDIF(FOUND_fribidi_utf8_to_unicode)
+	ENDIF (FRIBIDI_LIBRARY AND FRIBIDI_INCLUDE_DIR)
+
+	IF (FRIBIDI_FOUND)
+		IF (NOT FRIBIDI_FIND_QUIETLY)
+			MESSAGE(STATUS "Found FriBiDi: ${FRIBIDI_LIBRARY}")
+		ENDIF (NOT FRIBIDI_FIND_QUIETLY)
+		ELSE (FRIBIDI_FOUND)
+			IF (FRIBIDI_FIND_REQUIRED)
+				MESSAGE(FATAL_ERROR "Could not find FriBiDi library")
+			ENDIF (FRIBIDI_FIND_REQUIRED)
+	ENDIF (FRIBIDI_FOUND)
+endif()
+##################################33
+
+
+# Find Harfbuzz include and library directories.
+#
+# After successful discovery, this will set for inclusion where needed:
+# HARFBUZZ_INCLUDE_DIRS - containg the HarfBuzz headers
+# HARFBUZZ_LIBRARIES - containg the HarfBuzz library
+
+
+if(WIN32 AND NOT(MSYS OR CYGWIN OR MINGW))
+	FIND_PATH(HARFBUZZ_INCLUDE_DIRS hb.h
+		PATH
+		${PROJECT_SOURCE_DIR}/win/include/harfbuzz
+	)
+	
+	if(32bit)
+		FIND_LIBRARY(HARFBUZZ_LIBRARIES libharfbuzz-0
+			PATH
+			${PROJECT_SOURCE_DIR}/win/lib/windows32
+		)
+	else(32bit)
+		FIND_LIBRARY(HARFBUZZ_LIBRARIES libharfbuzz-0
+			PATH
+			${PROJECT_SOURCE_DIR}/win/lib/windows64
+		)
+	endif(32bit)
+  
+	message("-- Found Harfbuzz: ${HARFBUZZ_LIBRARIES} ")
+ 
+else()
+	INCLUDE(FindPkgConfig)
+	pkg_check_modules(PC_HARFBUZZ REQUIRED harfbuzz>=0.9.20)
+
+
+	FIND_PATH(HARFBUZZ_INCLUDE_DIRS NAMES hb.h
+		HINTS ${PC_HARFBUZZ_INCLUDE_DIRS} ${PC_HARFBUZZ_INCLUDEDIR} )
+
+	FIND_LIBRARY(HARFBUZZ_LIBRARIES NAMES harfbuzz
+		HINTS ${PC_HARFBUZZ_LIBRARY_DIRS} ${PC_HARFBUZZ_LIBDIR} )
+
+	IF (PC_HARFBUZZ_FOUND)
+		MESSAGE(STATUS "Harfbuzz Library Found OK:  ${HARFBUZZ_LIBRARIES} ")
+		INCLUDE(FindPackageHandleStandardArgs)
+		FIND_PACKAGE_HANDLE_STANDARD_ARGS(HarfBuzz DEFAULT_MSG HARFBUZZ_INCLUDE_DIRS HARFBUZZ_LIBRARIES)
+	ELSE ()
+		MESSAGE(FATAL_ERROR "No harfbuzz Found")
+	ENDIF()
+endif(WIN32 AND NOT(MSYS OR CYGWIN OR MINGW))
+####################################

```

Comment:
Same as above. Really so much work for what is usually done with single-line macro in autoconf!